cmd/frontend/auth/user.go

```diff
@@ -270,6 +273,26 @@ func GetAndSaveUser(ctx context.Context, db database.DB, op GetAndSaveUserOp) (n
 		PrivateMetadata: map[string]any{"serviceType": acct.AccountSpec.ServiceType},
 	})
 
+	if op.SingleProvider && !extAcctSaved {
+		// If we're here, the user has already signed up before this request,
+		// but has no entry for this exact accountID in the user external accounts
+		// table yet.
+		// In single provider mode, we want to attach a new external account
+		// only if no other identity for the same user already exists.
+		other, err := db.UserExternalAccounts().List(ctx, database.ExternalAccountsListOptions{
```

Comment:
Nit, we have a variable for this:

```suggestion
		other, err := externalAccountsStore.List(ctx, database.ExternalAccountsListOptions{
```